### ISSUE TITLE
Pinning Atom to the Windows taskbar duplicates the icon

### ISSUE THREAD

**Issue by tabrezm** [atom/atom#7374]
Since 0.210.0, it appears that Atom gets two icons instead of just one if Atom is pinned to the Windows taskbar. Clicking the pinned icon will open another icon that represents the running app. I would expect the first instance of Atom to reuse the pinned icon rather than adding a new one.

**Comment by batjko** [comment]
Your old pinned link won't be working then.
Could you try and remove it, then pin your new icon (when Atom is open)?

Did you previously install Atom via Chocolatey by any chance, or perhaps had other instances that were installed differently than with the standard installer in the past?

**Comment by tabrezm** [comment]
Yep, I tried actually pinning the icon for the active window, and pinning it adds a second icon to the taskbar immediately. I've only ever installed Atom using the official downloads, and upgraded using the in-app functionality. Nothing funky with Chocolatey or building from source.

**Comment by rjfmachado** [comment]
Seeing the same behavior with Windows 10 build 10130.

**Comment by jonegerton** [comment]
Also seeing this on Windows 10 build 10158.

It happens both ways,:
* Pinning the taskbar icon from an already running instance duplicates the icon.
* Opening Atom from a pinned icon (with no instances of atom running) also duplicates the icon.

This is on the above windows 10 build in a VirtualBox v4.3.28 with clean install of Atom v1.0.0

**Comment by batjko** [comment]
Thanks for adding this info, @jonegerton.
It's clear by now that taskbar pinning is handled different in Win10 and Squirrel may have to adapt.
Not sure when that happens though.

**Comment by jonegerton** [comment]
@batjko: Seems like it, although note that in some circumstances it doesn't work properly on 8.1 either as I've described.

**Comment by paulcbetts** [comment]
I'm going to be working on this bug soon

**Comment by tabrezm** [comment]
I noticed that on build-to-build upgrades of Win10, Atom disappears entirely from the Start menu. Typing 'atom' only results in a shortcut to 'atom.cmd' at "%localappdata%\atom\bin\atom.cmd".

**Comment by Phenome** [comment]
Can reproduce on newer Windows 10 builds... >= 10158, possibly even on older builds.

It *did* work on Windows 10... Been using W10 since they started the preview program. And I only found out about Atom after that... So I've been using atom pinned the whole time. And it worked.

It only stopped working when after a W10 build update, it was no longer pinned, so I opened it again and repinned. That's when it started. I assumed it was a Atom 1.0 problem, since it was coincidentally around the same time. I don't use atom every day.

But reading this issue then perhaps it really is a W10 thing. I tried changing the shortcut target to atom.cmd, and other configurations, but to no avail. There are always two icons on the taskbar if it's pinned and running.

**Comment by warpdesign** [comment]
+1
Hope this pull request fixes the problem.

**Comment by cristianvuolo** [comment]
any solution?

**Comment by paulcbetts** [comment]
@cristianvuolo This should be fixed once #7877 lands

**Comment by Zireael07** [comment]
In my case, there is the 'missing icon' graphic on the taskbar, not duplicates. Win 7 64 bit.

**Comment by paulcbetts** [comment]
@Zireael07 This is a different bug in Squirrel.Windows that I am going to fix today

**Comment by Apparate** [comment]
Is this fixed already? I am still getting two icon in my taskbar.

Currently using Windows 10 Pro.

**Comment by 50Wliu** [comment]
No, it's not yet fixed.  Please follow #7877 for more details.

**Comment by tinymins** [comment]
There's still two icons on my task bar.
Win10 x64, ATOM 1.0.10

**Comment by nickfmc** [comment]
I think this needs to be reopened - Updating to version 1.0.12 on Windows 10 removed my pinned icon, upon re-adding the pinned icon after an update I am seeing this behavior.

**Comment by jonathandelgado** [comment]
Yeah, +1 for reopen, #7877 did not fix this issue.

**Comment by tinymins** [comment]
Did not fixed yet.
WIN10x64, Atom 1.0.13

**Comment by Stanzilla** [comment]
@paulcbetts Can you do something about the pinned icon disappearing from the taskbar with each update?

**Comment by paulcbetts** [comment]
This is failing because of an indentation fuckup:

https://github.com/atom/atom/blob/f56dde9c91e9507258bc9b4f51e9df0508c3bc0e/src/browser/main.coffee#L20

I'll make a PR for this.

**Comment by Stanzilla** [comment]
Made a new one then https://github.com/atom/atom/issues/8858

**Comment by JSn1nj4** [comment]
I just started having this issue a week ago. I'm currently running Atom 1.0.15 on Windows 10. I've had 10 on my laptop for a few weeks now, and Atom has worked normally until last week (Thursday, if I remember correctly). Sorry for waiting this long to report it.

![atom-properties](https://cloud.githubusercontent.com/assets/5084820/9995666/5c864f26-6051-11e5-9c52-b5c032ff107d.png)

**Comment by paulcbetts** [comment]
@JSn1nj4 Yep, welcome to the nightmare that is Pinning, where the shortcut that Explorer creates is not at all related to the one that gets put in the Start Menu or anywhere else.

**Comment by nathansobo** [comment]
@paulcbetts Will fixing that indentation problem solve this?

**Comment by paulcbetts** [comment]
@nathansobo Yep

**Comment by JSn1nj4** [comment]
Thanks @paulcbetts. Looking forward to the next update.

**Comment by nathansobo** [comment]
This is fixed in 1.0.16.

**Comment by batjko** [comment]
:+1:

**Comment by jmlv** [comment]
:+1:

**Comment by mikeytusa** [comment]
Fixed.

**Comment by JSn1nj4** [comment]
Thanks a ton guys! Pinning works again.
:+1:

**Comment by tinymins** [comment]
👍

**Comment by CavemanNinja** [comment]
I'm still experiencing this issue in 1.2.3

**Comment by gregor3000** [comment]
This is still an issue (win10). Editors like Sublime, Notepad++ etc do not have this issue. Also most other old (pre-win10) programs work just fine.

**Comment by garygreen** [comment]
Sorry but this is still an issue - using 1.15.0, not fixed yet so this should be re-opened.

**Comment by mspiezio** [comment]
This is still an issue.  Win10 x64  Atom 1.15.0
Re-Open, or create a new ticket?

**Comment by jonathandelgado** [comment]
I can't reproduce on Win10 x64, could you try unpinning your current shortcut then pinning the new one? Maybe your paths didn't update with Atom.

**Comment by mspiezio** [comment]
I tried several unpinning / pinning scenarios previously with the shortcut, from the Start menu, the actual exe.  I also tried an uninstall, remove my .atom directory and re-install.  None of those worked.

**However, I found a pinning scenario that did work.**  Make sure it's unpinned from the taskbar, open Atom from Windows menu, with Atom open, right-click its icon in the taskbar, and there is an option to pin.  Selecting that pin method worked.

Just to assure myself I'm not crazy I went back to my previous pinning scenarios and those still didn't work... so at least it's consistent in my environment.

**Comment by jonathandelgado** [comment]
Yeah, that seems consistent with that I was thinking. The paths aren't getting updated correctly during an upgrade for existing shortcuts; this still occurs for me every once and a while. It isn't an issue with pinning, though.

**Comment by isherwood** [comment]
@mspiezio What does "from Windows menu" mean? How else would one open it? At any rate, I couldn't get your suggestion to work. I'm not sure how it's different from all the other things I've tried.

Please fix this, Atom team. I can't even tell you how often I've clicked the wrong icon and opened a second instance of Atom.

**Comment by warpdesign** [comment]
@isherwood Actually you have several ways to open an app in Windows 10:
 - from the start menu
 - from the taskbar (if you previously pinned the app)
 - from the explorer, by clicking on the app's icon
 - from the command line (I would guess the behavior would be the same that clicking on the app)

I can even think of a fith one: if you had associated some file types to Atom and you click on this type of file.

**Comment by isherwood** [comment]
Yeah, I've used Windows before, too. Not many folks go looking for executables or a command prompt during their workflow, though, and I assume that the program should be un-pinned first. Just looking for clarification on the phrase. I assume it means "Start menu".

**Comment by isherwood** [comment]
Problem remains in 1.17.0 ia32.

**Comment by cameronkerrnz** [comment]
Here are my findings, installing latest Atom release (as of 2017-10-21) on Windows 10 Enterprise 1703.

*If* you right click on the Atom menu in the Start menu and select Pin to Taskbar, you end up with a duplicate icon when the program lauches.

*Instead*, what you you should do [to avoid the duplicate icon] is to launch Atom and then right click on the resulting taskbar icon and select Pin to Taskbar. Then, if you close Atom, the icon will remain (perhaps it will disappear after an upgrade; I'm not sure), but when you launch it you will remain with one taskbar icon for Atom.

I suspect what a lot of people would be doing is right-clicking on the Start of Desktop menu and selecting Pin to Taskbar, in which case the resulting link points to

C:\Users\kerca54p\AppData\Local\atom\atom.exe

*but* that appears to be a wrapper, and is not what is eventually running; the thing that is running is

C:\Users\kerca54p\AppData\Local\atom\app-1.21.1\atom.exe

Hope that makes it clearer for people what is happening.

**Comment by isherwood** [comment]
Thank you @cameronkerrnz, but this has been suggested before. It results in an immediate duplicate icon for me.

**Comment by isherwood** [comment]
Thanks, but that shortcut doesn't exist for my user. Also, there's some confusion about which .exe to use. See https://github.com/atom/atom/issues/5055. I have at least 4, counting older version folders.

**Comment by phpcitizen** [comment]
isherwood,

I am using the portable version (.zip) for that reason.
